funding/manager.go

```diff
@@ -2441,11 +2452,10 @@ type confirmedChannel struct {
 }
 
 // fundingTimeout is called when callers of waitForFundingWithTimeout receive
-// an ErrConfirmationTimeout. It is used to clean-up channel state and mark the
-// channel as closed. The error is only returned for the responder of the
-// channel flow.
+// an ErrConfirmationTimeout or errFundingInputSpent error. It is used to
+// clean-up channel state and mark the channel as closed.
 func (f *Manager) fundingTimeout(c *channeldb.OpenChannel,
-	pendingID [32]byte) error {
+	pendingID [32]byte, fundingErr error) error {
```

Comment:
Using the general error interface here means that we can _technically_ pass anything in here. Seems like it would make sense to either make a specific error type or use an enum to restrict use?